core/templates/pages/exploration-editor-page/services/exploration-warnings.service.ts

```diff
@@ -235,6 +237,39 @@ export class ExplorationWarningsService {
     return indexes;
   }
 
+  _getStatesWithInvalidRedirection(
+      initStateId: string, links: GraphLink[]
+  ): string[] {
+    let results = [];
+    let states = this.explorationStatesService.getStates();
+    let bfsStateList = this.computeGraphService.computeBfsTraversalOfStates(
+      initStateId, states, initStateId);
+    let seen: Record<string, boolean> = {};
+    bfsStateList.forEach((stateName) => {
+      // Go through all states, taking in the dest and source
+      // calculate distance for each, if invalid push in results.
+      seen[stateName] = true;
+      let state = states.getState(stateName);
+      if (state && state.interaction) {
+        if (state.interaction.defaultOutcome) {
+          let defaultDest = state.interaction.defaultOutcome.dest;
+          if (defaultDest !== stateName && seen[defaultDest] &&
+            !this.redirectionIsValid(defaultDest, stateName, links)) {
+            results.push(stateName);
+          }
+        }
+        let answerGroups = state.interaction.answerGroups;
+        for (let i = 0; i < answerGroups.length; i++) {
+          let dest = answerGroups[i].outcome.dest;
```

Comment:
Shouldn't we also check for the stuck state outcome?

Reply:
I did consider that. Actually, considering the dest_if_stuck link in the path didn't really make sense because it's a remedial path that is to be used when the learner gets really stuck on some state (fulfill to-be defined time and incorrect responses metrics). Counting them for the present purpose didn't actually really make sense to me. What do you think?

Reply:
Makes sense 👍